### PR TITLE
Production ready s4 proxy

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:
- Add ProxyURL column to Webhooks
- Upon Application Update, check if a proxyURL needs to be added on the webhook that is being created. This info is determined by the ORD Mapping Configuration array.
- When calling ORD webhooks use the proxyURL if it is present and enrich the headers using the ORD Mapping Configuration data.

**Related issue(s)**

- ...

**Pull Request status**

- [ ] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/open_resource_discovery/service.go [hunk 1]
```diff
@@ -1774,8 +1783,43 @@ func (s *Service) processWebhookAndDocuments(ctx context.Context, cfg MetricsCon
...
+		Application: requestobject.Application{BaseURL: str.PtrStrToStr(appBaseURL)},
+		Headers:     http.Header{},
+	}
+
+	if webhook.HeaderTemplate != nil {
+		headers, err := ordRequestObject.ParseHeadersTemplate(webhook.HeaderTemplate)
+		if err != nil {
+			return err
+		}
+		ordRequestObject.Headers = headers
```
Comment (kaloyanspiridonov) [inline components/director/internal/open_resource_discovery/service.go:1818]: i prefer appending of headers instead of overriding

**Review by kaloyanspiridonov — CHANGES_REQUESTED** [review]

File: components/director/internal/domain/webhook/entity.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ type Entity struct {
 	Type                  string         `db:"type"`
 	Mode                  sql.NullString `db:"mode"`
 	URL                   sql.NullString `db:"url"`
+	ProxyURL              sql.NullString `db:"proxy_url"`
```
[the remark this answers is not in the document]
Comment (alextargov) [reply components/director/internal/domain/webhook/entity.go:22]: We will need this, yes. When this go to dev, we should execute a one time SQL script to populate this column.

File: installation/resources/compass-overrides-local.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ global:
   director:
     subscription:
       subscriptionProviderLabelKey: "subscriptionProviderId"
-    ordWebhookMappings: '[{ "OrdUrlPath": "/.well-known/open-resource-discovery", "Type": "SAP temp1", "PpmsProductVersions": ["12345"], "SubdomainSuffix": "" }]'
+    ordWebhookMappings: '[{ "OrdUrlPath": "/.well-known/open-resource-discovery", "Type": "SAP temp1", "PpmsProductVersions": ["12345"], "SubdomainSuffix": "" }, { "ProxyURL": "http://compass-external-services-mock.compass-system.svc.cluster.local:8090", "ProxyHeaderTemplate": "{\"target_host\": [\"{{.Application.BaseURL}}\"] }", "OrdUrlPath": "/proxy", "SubdomainSuffix": "-api", "Type": "SAP Proxy Template" }]'
```
[the remark this answers is not in the document]
Comment (alextargov) [reply installation/resources/compass-overrides-local.yaml:28]: We can, but this is mainly for the e2e tests executed locally.

File: tests/ord-aggregator/tests/handler_test.go [hunk 4]
```diff
@@ -141,46 +136,682 @@ var (
 func TestORDAggregator(stdT *testing.T) {
 	t := testingx.NewT(stdT)
 
-	basicORDConfigSecurity := &fixtures.ORDConfigSecurity{
-		Username: testConfig.BasicUsername,
-		Password: testConfig.BasicPassword,
-	}
+	//basicORDConfigSecurity := &fixtures.ORDConfigSecurity{
```
[the remark this answers is not in the document]
Comment (alextargov) [reply tests/ord-aggregator/tests/handler_test.go:139]: Oops

**Review by kaloyanspiridonov — DISMISSED** [review]

**Review by NickyMateev — APPROVED** [review]

**Review by NickyMateev — APPROVED** [review]

**Pull request merged**